test/addIndex.js

```diff
@@ -0,0 +1,74 @@
+var assert = require('assert');
+
+var R = require('..');
+
+
+describe('addIndex', function() {
+  describe('works with unary functions like `map`', function() {
+    var times2 = function(x) {return x * 2;};
+    var addIndexParam = function(x, idx) {return x + idx;};
+    var squareEnds = function(x, idx, list) {
+      return (idx === 0 || idx === list.length - 1) ? x * x : x;
+    };
+    var mapIndexed = R.addIndex(R.map);
+
+    it('working just like a normal map', function() {
```

Comment:
It's hard to decide if it's the source code or the output that we want to look correct.

In other words, ooga ooga.